### ISSUE TITLE
SSH Hangs with Key Auth 

### ISSUE THREAD

**Issue by BobbyRyterski** [test-kitchen/kitchen-openstack#101]
With some configurations using SSH key authentication, kitchen-openstack can hang while checking for SSH to be available. This is likely related to issue #91 and maybe #77.

A configuration that fails:

```yaml
driver:
  name: openstack
  username: cloud-user
  openstack_auth_url: http://openstack/v2.0/tokens
  openstack_username: bobby
  openstack_api_key: password
  key_name: my_key
  private_key_path: /path/to/my_key.pem
  floating_ip_pool: pool0
  network_ref: network0
  security_groups:
    - security_group0
```

With debug logging turned on, I can see that the SSH check does not use the username or private key.

```
Waiting for server to be ready...
D      [SSH] opening connection to root@10.190.188.73<{:user_known_hosts_file=>"/dev/null", :paranoid=>false, :port=>22, :compression=>"zlib", :compression_level=>6, :keepalive=>true, :keepalive_interval=>60, :timeout=>15}>
D      [SSH] connection failed (#<Timeout::Error: execution expired>)
       Waiting for SSH service on 10.190.188.73:22, retrying in 3 seconds
D      [SSH] opening connection to root@10.190.188.73<{:user_known_hosts_file=>"/dev/null", :paranoid=>false, :port=>22, :compression=>"zlib", :compression_level=>6, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :user=>"root"}>
D      [SSH] connection failed (#<Errno::ECONNREFUSED: Connection refused - connect(2) for "10.190.188.73" port 22>)
       Waiting for SSH service on 10.190.188.73:22, retrying in 3 seconds
D      [SSH] opening connection to root@10.190.188.73<{:user_known_hosts_file=>"/dev/null", :paranoid=>false, :port=>22, :compression=>"zlib", :compression_level=>6, :keepalive=>true, :keepalive_interval=>60, :timeout=>15, :user=>"root"}>
D      [SSH] connection failed (#<Errno::ECONNREFUSED: Connection refused - connect(2) for "10.190.188.73" port 22>)
```

With v1.8.1 I was able to work around this by setting `ssh_key` to the same value as `private_key_path`. Since v2.0.0 uses transport, I got this to work:

```yaml
transport:
  username: cloud-user
  ssh_key: /path/to/my_key.pem
```

Is this intended or can we tweak `#wait_for_server` to set up SSH?

```ruby
def wait_for_server(state)
  # probably want to add these conditionally
  state[:ssh_key] = config[:private_key_path]
  state[:username] = config[:username]
  
  state[:hostname] = get_ip(state)
# ...
```

**Comment by jjasghar** [comment]
Ah so we need to add to the `README.md`

```yaml
transport:
  username: cloud-user
  ssh_key: /path/to/my_key.pem
```

So this is explicit. We should probably pull the code related to the old way not relying on the transport layer.

**Comment by BobbyRyterski** [comment]
Works for me. Do you want to add it or should I send a PR?

Also, props on the fast reply.

**Comment by jjasghar** [comment]
I'm a horrible writer, go ahead. I think around where we mention the transport already this should fit right in.

Thanks, I try to make sure that even if i can't help resolve an issue i at least acknowledge it and give some type of feedback and timeline.  Only successful way to run an OpenSource project ;).

**Comment by jgrill** [comment]
> With v1.8.1 I was able to work around this by setting ssh_key to the same value as private_key_path. Since v2.0.0 uses transport, I got this to work...

This fixes my problem #91with both 1.8.1 and 2.x
